plugins/nodes/src/js/components/NodesTable.tsx

```diff
@@ -56,7 +57,8 @@ export default class NodesTable extends React.Component<
     this.state = {
       data: [],
```

Comment:
Works great. 

As for the implementation i think that using a separate flag that's strongly coupled to data might be a little brittle. 

I'd prefer to go in the direction of what's suggested in this [article](https://medium.com/javascript-inside/slaying-a-ui-antipattern-in-react-64a3b98242c).

The fully fledged version might be something for later,  could set data to `null` initially and decide on what to render by the difference of `null` and `[]` although that might require a little type-mangling.

What do you think?

Reply:
To me, both doing `Node [] | null` and an additional state property is sub-optimal.